File: 0035-search-insert-position/0035-search-insert-position.cpp
// Using Binary Search Approach...
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int beg = 0;
        int end = nums.size()-1;
        while(beg <= end){
            int mid = (beg + end)/2;
            if(target > nums[mid]){
                beg = mid + 1;
            }else if(target < nums[mid]){
                end = mid - 1;
            }else{
                return mid;
            }
        }
        return beg;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.java
// The insert position is the index which element’s value is greater or equal with target.
// So we scan linearly the array will find the answer.
class Solution {
    public int searchInsert(int[] nums, int target) {
        //Traverse all elements through the loop...
        for(int i = 0; i < nums.length; i++){
            if(nums[i] >= target)   return i;
        }
    return nums.length;
    }
}